TYPO3.Flow/Documentation/TheDefinitiveGuide/PartV/CodingGuideLines/PHP.rst

```diff
@@ -375,113 +370,36 @@ These are the rules for naming files:
 * Each file must contain only one class or interface
 * Names of files containing code for unit tests must be the same as the class which is
   tested, appended with "Test.php".
-* Files are placed in a directory structure representing the namespace structure.
+* Files are placed in a directory structure representing the namespace structure. You may
+  use PSR-0 ro PSR-4 autoloading as you like. We generally use PSR-4.
 
 *File naming in Flow*
 
-``TYPO3.TemplateEngine/Classes/TYPO3/TemplateEngine/TemplateEngineInterface.php``
-  Contains the interface ``\TYPO3\TemplateEngine\TemplateEngineInterface`` which is part
-  of the package *TYPO3.TemplateEngine*
+``Neos.TemplateEngine/Classes/TemplateEngineInterface.php``
+  Contains the interface ``\Neos\TemplateEngine\TemplateEngineInterface`` which is part
+  of the package *Neos.TemplateEngine*
 
-``TYPO3.Flow/Classes/TYPO3/Flow/Error/RuntimeException.php``
-  Contains the ``\TYPO3\Flow\Error\RuntimeException`` being a part of the package
-  *TYPO3.Flow*
+``Neos.Flow/Classes/Error/RuntimeException.php``
+  Contains the ``\Neos\Flow\Error\Messages\RuntimeException`` being a part of the package
+  *Neos.Flow*
 
-``Acme.DataAccess/Classes/Acme/DataAccess/CustomQuery.php``
+``Acme.DataAccess/Classes/CustomQuery.php``
   Contains class ``\Acme\DataAccess\CustomQuery`` which is part of the package
   *Acme.DataAccess*
 
-``TYPO3.Flow/Tests/Unit/Package/PackageManagerTest.php``
-    Contains the class ``\TYPO3\Flow\\Tests\Unit\Package\PackageManagerTest`` which
+``Neos.Flow/Tests/Unit/Package/PackageManagerTest.php``
+    Contains the class ``\Neos\Flow\Tests\Unit\Package\PackageManagerTest`` which
     is a PHPUnit testcase for ``Package\PackageManager``.
 
 
 PHP code formatting
 ===================
 
-Strings
```

Comment:
The part on string concatenation must stay, turns out PSR-2 doesn't specify this.